phase2/src/lib.rs

```diff
@@ -14,31 +15,36 @@ cfg_if! {
         use wasm_bindgen::prelude::*;
         use itertools::Itertools;
         use parameters::MPCParameters;
-        use std::io::{
-            Read,
-            Write,
-        };
+        use zexe_algebra::{Bls12_377, BW6_761, PairingEngine};
+        use setup_utils::{ get_rng, user_system_randomness };
 
         macro_rules! log {
             ($($t:tt)*) => (web_sys::console::log_1(&format_args!($($t)*).to_string().into()))
         }
 
         #[wasm_bindgen]
-        pub fn contribute(params: Vec<u8>) -> Result<Vec<u8>, JsValue> {
+        pub fn contribute(is_inner: bool, params: Vec<u8>) -> Result<Vec<u8>, JsValue> {
             console_error_panic_hook::set_once();
 
             log!("Initializing phase2");
-            let mut rng = &mut rand::XorShiftRng::new_unseeded(); // TODO: change this unsafe unseeded random (!)
-            let mut params = MPCParameters::read(&*params, true).expect("unable to read params");
+            let res = match is_inner {
+                true => contribute_challenge(&mut MPCParameters::<Bls12_377>::read(&*params).unwrap()),
+                false => contribute_challenge(&mut MPCParameters::<BW6_761>::read(&*params).unwrap()),
+            };
 
+            Ok(res)
+        }
+
+        fn contribute_challenge<E: PairingEngine>(params: &mut MPCParameters<E>) -> Vec<u8> {
```

Comment:
Can you clarify why this function was added? My understanding was that the previous version was already working in WASM

Reply:
Well... previous version had build error so I refactored it like this.
Is this phase2 param contribute wasm code, right?